hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/DecayRpcSchedulerUtil.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.  The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.hadoop.hdds.utils;
+
+import com.google.common.base.Strings;
+import org.apache.hadoop.metrics2.MetricsInfo;
+import org.apache.hadoop.metrics2.MetricsRecord;
+import org.apache.hadoop.metrics2.MetricsTag;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Helper functions for DecayRpcScheduler
+ * metrics for Prometheus.
+ */
+public final class DecayRpcSchedulerUtil {
+
+  private DecayRpcSchedulerUtil() {
+  }
+
+  private static final MetricsInfo USERNAME_INFO = new MetricsInfo() {
+    @Override
+    public String name() {
+      return "username";
+    }
+
+    @Override
+    public String description() {
+      return "caller username";
+    }
+  };
+
+  /**
+   * For Decay_Rpc_Scheduler, the metric name is in format
+   * "Caller(<callers_username>).Volume"
+   * or
+   * "Caller(<callers_username>).Priority"
+   * Split it and return the metric.
+   * @param recordName
+   * @param metricName "Caller(xyz).Volume" or "Caller(xyz).Priority"
+   * @return "Volume" or "Priority"
+   */
+  public static String splitMetricNameIfNeeded(String recordName,
```

Comment:
Please add unit tests for these methods to make sure the string processing is what we expect.

Reply:
@kerneltime I added the unit tests you requested.